yt_dlp/extractor/gedidigital.py

```diff
@@ -94,9 +101,49 @@ class GediDigitalIE(InfoExtractor):
         'only_matching': True,
     }]
 
+    @staticmethod
+    def _sanitize_urls(urls):
+        # add protocol if missing
+        for i, e in enumerate(urls):
+            if e.startswith('//'):
+                urls[i] = 'https:%s' % e
+        # clean iframes urls
+        for i, e in enumerate(urls):
+            urls[i] = urljoin(base_url(e), url_basename(e))
+        return urls
+
+    @staticmethod
+    def _extract_urls(webpage):
+        entries = [
+            mobj.group('eurl')
+            for mobj in re.finditer(r'''(?x)
+            (?:
+                data-frame-src=|
+                <iframe[^\n]+src=
+            )
+            (["'])(?P<eurl>%s)\1''' % GediDigitalIE._VALID_URL, webpage)]
+        return GediDigitalIE._sanitize_urls(entries)
+
+    @staticmethod
+    def _extract_url(webpage):
+        urls = GediDigitalIE._extract_urls(webpage)
+        return urls[0] if urls else None
+
+    @staticmethod
+    def _clean_formats(formats):
+        format_urls = set()
+        clean_formats = []
+        for f in formats:
+            if f['url'] not in format_urls:
+                if f.get('audio_ext') != 'none' and not f.get('acodec'):
+                    continue
```

Comment:
could you explain the intention behind this?

Reply:
remove duplicate audio formats generated by the _extract_m3u8_formats method.


```
audio-hls-vod-1@48000-64000-Audio_64 mp4 audio only |       m3u8_n |                   unknown        
audio-hls-vod-1@48000-80000-Audio_80 mp4 audio only |       m3u8_n |                   unknown        
audio-hls-vod-2@48000-96000-Audio_96 mp4 audio only |       m3u8_n |                   unknown        
audio-hls-vod-100                    mp4 audio only |  100k m3u8_n |                   mp4a.40.2 100k 
audio-hls-vod-116                    mp4 audio only |  116k m3u8_n |                   mp4a.40.2 116k 
audio-hls-vod-133                    mp4 audio only |  133k m3u8_n |                   mp4a.40.2 133k 
```

as you can see there are 6 audio formats but only 3 in reality. I removed the 3 formats that have a "weird" `format_id` and missing `acodec` / `bitrate` information

Reply:
You are not checking whether the "more detailed" format exists though. I am gonna assume u are sure that such a format always exists

Reply:
yup.

it's the _extract_m3u8_formats method that IMHO is designed with video streams and for only audio formats it does not check well and create duplicates.